src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.cs

```diff
@@ -354,7 +354,7 @@ private static bool IsOnObsoleteMemberChain(ISymbol symbol, WellKnownTypeProvide
             while (symbol != null)
             {
                 allAttributes.AddRange(symbol.GetAttributes());
-                symbol = symbol is IMethodSymbol method && method.AssociatedSymbol != null
+                symbol = symbol is IMethodSymbol { AssociatedSymbol: not null } method
```

Comment:
This is a manual change. The codefix produces invalid code.

Reply:
Has the bug been filed?

Reply:
Oh, this is https://github.com/dotnet/roslyn/issues/51691.